### PR TITLE
feat: Add user guide for compaction offload

### LINKED ISSUE #156: Add how to use remote compaction docs

### PR DESCRIPTION
**Pull request by LeslieKid**
Close #156

### CODE REVIEW

File: content/en/docs/user-guide/cluster_deployment/with_meta.md [hunk 1]
```diff
@@ -187,9 +187,27 @@ timeout = "5s"
...
 ```
 
+### Compaction Offload
+
+Compaction offload is also supported in `WithMeta` mode. To enable compaction offload, the compaction mode of HoraeDB Server should be configured as remote compaction:
+
+```toml
+[analytic.compaction_mode]
+compaction_mode = "Offload"
+node_picker = "Remote"
```
Comment (jiacai2050) [inline content/en/docs/user-guide/cluster_deployment/with_meta.md:197]: `endpoint` is also required?

File: content/en/docs/user-guide/cluster_deployment/with_meta.md [hunk 2]
```diff
@@ -187,9 +187,27 @@ timeout = "5s"
 server_addrs = ['http://{HoraeMetaAddr}:2379']
 ```
 
+### Compaction Offload
+
+Compaction offload is also supported in `WithMeta` mode. To enable compaction offload, the compaction mode of HoraeDB Server should be configured as remote compaction:
```
Comment (jiacai2050) [inline content/en/docs/user-guide/cluster_deployment/with_meta.md:192]: IMO, the current implementation doesn't depend on horaemeta?

File: content/en/docs/user-guide/cluster_deployment/with_meta.md [hunk 3]
```diff
@@ -187,9 +187,34 @@ timeout = "5s"
...
+  - When the `Local` is setted, the local compaction task would be offloaded to the specific remote compaction server, which decided by `endpoint`.
+- `endpoint`: The endpoint, in the form `addr:port`, indicating the _grpc port_ of the remote compaction server.
+
+Here is an example for it:
+
+```toml
+[analytic.compaction_mode]
+compaction_mode = "Offload"
+node_picker = "Local"
+endpoint = "{RemoteCompactionServerAddr}:{RemoteCompactionServerGrpcPort}"
```
Comment (LeslieKid) [inline content/en/docs/user-guide/cluster_deployment/with_meta.md:204]: The remote node picker is not supported without horaemeta. Currently, we need to use local node picker and specify the `endpoint`.

File: content/en/docs/user-guide/cluster_deployment/with_meta.md [hunk 4]
```diff
@@ -187,9 +187,34 @@ timeout = "5s"
...
+```
+
+A Compaction Server, responsible for executing the compaction task, is also needed. The configuration of Compaction Server is similar to the HoraeDB Server, except that the node type (default `HoraeDB`) should be set to `CompactionServer`:
```
[the remark this answers is not in the document]
Comment (LeslieKid) [reply content/en/docs/user-guide/cluster_deployment/with_meta.md:207]: > This is required?

Right. It is not required. It is used to help horaemeta distinguish the horaedb node and compaction node, which is not supported now.

File: content/en/docs/user-guide/cluster_deployment/with_meta.md [hunk 5]
```diff
@@ -187,9 +187,28 @@ timeout = "5s"
...
+Here is an example for it:
+
+```toml
+[analytic.compaction_mode]
+compaction_mode = "Offload"
+node_picker = "Local"
+endpoint = "{RemoteCompactionServerAddr}:{RemoteCompactionServerGrpcPort}"
+```
+
+A Compaction Server, responsible for executing the compaction task, is also needed. The configuration of Compaction Server is similar to the HoraeDB Server.
```
Comment (jiacai2050) [inline content/en/docs/user-guide/cluster_deployment/with_meta.md:207]:
```suggestion
A Compaction Server, responsible for executing the compaction task, is also needed. Currently `horaedb-server` will act as this role, in the future we can move it to an independent service.
```

**Review by jiacai2050 — APPROVED** [review]
Thanks, LGTM.

**Comment by jiacai2050** [comment]
@LeslieKid Welcome to join [our community](https://horaedb.apache.org/community/), we are implementing a [new engine](https://github.com/apache/horaedb/blob/main/docs/rfcs/20240827-metric-engine.md), and lots of work waiting to be done.

Or just ping me if you are already in our Discord/Dingtalk group.

**Pull request merged**